apps/payments/app/[locale]/(hosted)/paymentRequest/stripe/page.tsx

```
@@ -119,10 +117,9 @@ export default async function Card(props: {
     >
       <NextIntlClientProvider messages={stripeMessages}>
         <StripeHost
-          clientSecret={paymentIntent.client_secret as string | undefined}
+          clientSecret={paymentIntent.data.clientSecret as string | undefined}
```

In case clientSecret ends up being undefined, I'd expect the API request to fail, shall we remove this `as`?

---

the payment intent api can actually return a null client_secret according to their signature. It is not clear in which use cases this would happen, so I would avoid forcing our API to fail (the payment intent could have been created anyways). Maybe the frontend could error in this case?

---

Interesting, I assumed it would either return the client_secret or error. I agree with you that we can throw an error on the frontend in that case

---

[2da1afe](https://github.com/ogcio/payments-building-block/pull/34/commits/2da1afe99fd72b857497ddba0c5659a2a96bcba4)